### PR TITLE
HDDS-1856. Make required changes for Non-HA to use new HA code in OM.

### PR DESCRIPTION
**Pull request by bharatviswa504**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 75 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 25 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 734 | trunk passed |
| +1 | compile | 361 | trunk passed |
| +1 | checkstyle | 71 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 898 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 167 | trunk passed |
| 0 | spotbugs | 434 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 632 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 574 | the patch passed |
| +1 | compile | 376 | the patch passed |
| +1 | javac | 376 | the patch passed |
| -0 | checkstyle | 41 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 726 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 161 | the patch passed |
| +1 | findbugs | 800 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 331 | hadoop-hdds in the patch failed. |
| -1 | unit | 2061 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 8252 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.block.TestBlockManager |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.web.client.TestBuckets |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1174 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux aeb69cfe5e59 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 2fe450c |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/1/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/1/testReport/ |
| Max. process+thread count | 4080 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 48 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 2 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 25 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 623 | trunk passed |
| +1 | compile | 368 | trunk passed |
| +1 | checkstyle | 78 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 910 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 220 | trunk passed |
| 0 | spotbugs | 899 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 1385 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 1024 | the patch passed |
| +1 | compile | 350 | the patch passed |
| +1 | javac | 350 | the patch passed |
| +1 | checkstyle | 83 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 681 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 162 | the patch passed |
| +1 | findbugs | 638 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 283 | hadoop-hdds in the patch failed. |
| -1 | unit | 1640 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 48 | The patch does not generate ASF License warnings. |
| | | 8918 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1174 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 3f257c0ebf60 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 611718f |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/2/testReport/ |
| Max. process+thread count | 4205 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java [hunk 1]
```diff
@@ -169,15 +163,27 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
...
+          CreateBucketResponse.newBuilder().build());
+      omClientResponse = new OMBucketCreateResponse(omBucketInfo,
+          omResponse.build());
     } catch (IOException ex) {
       exception = ex;
+      omClientResponse = new OMBucketCreateResponse(omBucketInfo,
+          createErrorOMResponse(omResponse, exception));
     } finally {
+      if (omClientResponse != null) {
+        omClientResponse.setFlushFuture(
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java]: Can the setFlushFuture operation be moved to the caller? I assume every request has to do this.

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/bucket/TestS3BucketRequest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+public class TestS3BucketRequest {
+  @Rule
+  public TemporaryFolder folder = new TemporaryFolder();
+
+  protected OzoneManager ozoneManager;
+  protected OMMetrics omMetrics;
+  protected OMMetadataManager omMetadataManager;
+  protected AuditLogger auditLogger;
+
+  // Just setting ozoneManagerDoubleBuffer which does no
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/bucket/TestS3BucketRequest.java]: incomplete comment?

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMKeyRequest.java [hunk 3]
```diff
@@ -82,6 +83,12 @@
   protected long scmBlockSize = 1000L;
   protected long dataSize;
 
+  // Just setting ozoneManagerDoubleBuffer which does no
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMKeyRequest.java]: incomplete comment?

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/multipart/TestS3MultipartRequest.java [hunk 4]
```diff
@@ -56,6 +57,12 @@
   protected OMMetadataManager omMetadataManager;
   protected AuditLogger auditLogger;
 
+  // Just setting ozoneManagerDoubleBuffer which does no
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/multipart/TestS3MultipartRequest.java]: same..

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/volume/TestOMVolumeRequest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+public class TestOMVolumeRequest {
+
+  @Rule
+  public TemporaryFolder folder = new TemporaryFolder();
+
+  protected OzoneManager ozoneManager;
+  protected OMMetrics omMetrics;
+  protected OMMetadataManager omMetadataManager;
+  protected AuditLogger auditLogger;
+  // Just setting ozoneManagerDoubleBuffer which does no
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/volume/TestOMVolumeRequest.java]: same

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java [hunk 6]
```diff
@@ -61,6 +63,10 @@
   private Queue<DoubleBufferEntry<OMClientResponse>> currentBuffer;
   private Queue<DoubleBufferEntry<OMClientResponse>> readyBuffer;
 
+
+  private Queue<CompletableFuture<Void>> currentFutureQueue;
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java]: Can you add a one-line comment for the new fields? 

Also we should probably add for the existing fields. I think some ASCII art description of how double buffer works will be helpful to future maintainers. However it's okay to file a follow up jira and do separately later. Don't need to do it for this commit.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java [hunk 7]
```diff
@@ -248,10 +291,20 @@ public long getFlushIterations() {
    * @param response
    * @param transactionIndex
    */
-  public synchronized void add(OMClientResponse response,
+  public synchronized CompletableFuture<Void> add(OMClientResponse response,
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java]: Can you return Optional<CompletableFuture<Void>> instead? We should avoid having nulls in code wherever possible.

**Comment by arp7** [comment]
Basically +1 from me. A few minor comments.

I thought about the potential synchronization issue @anuengineer pointed out offline and am not convinced it exists. Because we hold/release the lock when flushTransactions calls setReadyBuffer. So that is the synchronization point. All later access from the same thread should see the correct value of the futureQueue.

Couple of ways remove the ambiguity:
1. Make the queues volatile.
1. Return the queue from setReadyBuffer. Since the returned pointer was sampled with the lock held, the caller is guaranteed to see the correct value. This solution should also make findbugs happy.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java (hunk 7 above)
Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java]: This is for a temporary thing. Sooner, all OM uses HA, so this code will be removed in later point of time. And in HA case, we don't even use future. So, I think this should be okay, let me know you still want to use Optional here?

Comment (arp7) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java]: No that's fine. Leave it as it is.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java (hunk 6 above)
Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java]: Opened jira for this.
https://issues.apache.org/jira/browse/HDDS-1874

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/bucket/TestS3BucketRequest.java (hunk 2 above)
Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/bucket/TestS3BucketRequest.java]: Yes, thanks for catching it. Fixed it.

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMKeyRequest.java (hunk 3 above)
Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMKeyRequest.java]: fixed it.

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/volume/TestOMVolumeRequest.java (hunk 5 above)
Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/volume/TestOMVolumeRequest.java]: Fixed it.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 45 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 25 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 625 | trunk passed |
| +1 | compile | 363 | trunk passed |
| +1 | checkstyle | 83 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 848 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 150 | trunk passed |
| 0 | spotbugs | 419 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 614 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 561 | the patch passed |
| +1 | compile | 352 | the patch passed |
| +1 | javac | 352 | the patch passed |
| +1 | checkstyle | 71 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 617 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 153 | the patch passed |
| +1 | findbugs | 686 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 194 | hadoop-hdds in the patch failed. |
| -1 | unit | 1915 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 7473 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.ozoneimpl.TestOzoneContainer |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestContainerStateMachine |
|   | hadoop.ozone.client.rpc.TestHybridPipelineOnDatanode |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1174 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 4bbc9ab239dc 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 0f2dad6 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/4/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/4/testReport/ |
| Max. process+thread count | 3749 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 82 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 2 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 25 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 717 | trunk passed |
| +1 | compile | 374 | trunk passed |
| +1 | checkstyle | 73 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 889 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 161 | trunk passed |
| 0 | spotbugs | 413 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 603 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 559 | the patch passed |
| +1 | compile | 364 | the patch passed |
| +1 | javac | 364 | the patch passed |
| +1 | checkstyle | 81 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 640 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 152 | the patch passed |
| +1 | findbugs | 631 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 350 | hadoop-hdds in the patch failed. |
| -1 | unit | 3430 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 39 | The patch does not generate ASF License warnings. |
| | | 9290 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestHybridPipelineOnDatanode |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestDeleteContainerHandler |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestBlockOutputStream |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.Test2WayCommitInRatis |
|   | hadoop.ozone.client.rpc.TestContainerStateMachineFailures |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.ozone.web.client.TestOzoneClient |
|   | hadoop.hdds.scm.pipeline.TestPipelineClose |
|   | hadoop.ozone.TestContainerStateMachineIdempotency |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.container.common.transport.server.ratis.TestCSMMetrics |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1174 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ab84367207b3 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 0f2dad6 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/3/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/3/testReport/ |
| Max. process+thread count | 3684 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1174/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bharatviswa504** [comment]
Opened a jira HDDS-1872 for failure related to TestS3MultipartUploadAbortResponse.
Rest of the test failures are not related to this patch.
Thank You @arp7 for the review. I will commit this to the trunk.

**Pull request merged**